### PR TITLE
Extracting presenters to common module

### PR DESCRIPTION
**Pull request by MarcinMoskala**
The big idea behind everything was that I was trying to make everything simpler and make common parts common. I made a classic MVP with coroutines (which are supported in common modules). I simplified the whole flow. We do not depend on LiveData anymore (which are not multiplatform). Here is whole logic:

https://github.com/JetBrains/kotlinconf-app/tree/move_logic_to_common_module/common/src/main/kotlin/com/jetbrains/kotlinconf/presentation

Presenters:
https://github.com/JetBrains/kotlinconf-app/blob/move_logic_to_common_module/common/src/main/kotlin/com/jetbrains/kotlinconf/presentation/SessionDetailsPresenter.kt
https://github.com/JetBrains/kotlinconf-app/blob/move_logic_to_common_module/common/src/main/kotlin/com/jetbrains/kotlinconf/presentation/SessionListPresenter.kt

Fragments are also now shorter. Not only logic extraction, but also I made there some clean-up, so I extracted some functions:

https://github.com/JetBrains/kotlinconf-app/compare/master...move_logic_to_common_module#diff-1c835fab6828f5708bbbaa4c936e623c
https://github.com/JetBrains/kotlinconf-app/compare/master...move_logic_to_common_module#diff-3b58872ce1b1b2d1d2f571f3c577e2d7

There is still some logic in KotlinConf repository which is still huge and complex, but I believe that it is the next step. AFAIU @e5l is willing to move it to the common module and simplified it during that process. 

I also introduced a local flavour for local application testing. To use it, just start backend by `./gradlew :backend:run` and start the application with local flavour (it is now default, we can change it when we will have some real backend running).

### CODE REVIEW

File: .idea/vcs.xml [hunk 1]
```diff
@@ -2,6 +2,5 @@
 <project version="4">
   <component name="VcsDirectoryMappings">
     <mapping directory="" vcs="Git" />
-    <mapping directory="$PROJECT_DIR$" vcs="Git" />
```
[the remark this answers is not in the document]
Comment (MarcinMoskala) [reply .idea/vcs.xml]: Well, .idea is already in the project. I can add it, but it won't be easy to get rid of it either way.

File: common/src/main/kotlin/com/jetbrains/kotlinconf/presentation/SessionDetailsPresenter.kt [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        private val view: SessionDetailsView,
+        private val sessionId: String,
+        private val repository: KotlinConfDataRepository
+) {
+    private lateinit var session: SessionModel
+    private var isFavorite: Boolean by observable(false) { _, _, isFavorite ->
+        view.setIsFavorite(isFavorite)
+    }
+    private var rating: SessionRating? = null
+    private val onRefreshListener: ()->Unit = this::refreshDataFromRepo
```
[the remark this answers is not in the document]
Comment (MarcinMoskala) [reply common/src/main/kotlin/com/jetbrains/kotlinconf/presentation/SessionDetailsPresenter.kt]: To be honest, I don't really see it.

File: .idea/vcs.xml (hunk 1 above)
[the remark this answers is not in the document]
Comment (MarcinMoskala) [reply .idea/vcs.xml]: Well, this file was already committed. I am fine with adding `/.idea` to `.gitignore`, but I don't think it needs to be done in this branch.

**Comment by MarcinMoskala** [comment]
@jarekankowski I tried to make minimal PR, and this is what I end up with. I think it cannot be easily split. Previous architecture depends strongly on LiveData and I had to make all these changes to make common parts independent of this library. 

DI is suggested in this architecture. This is why I made dependency injection via constructor: to allow easy introduction of DI framework like Dagger. To support it even better, we can switch later to view passing in `onCreate` method. Though I think DI should be introduced in another PR.

**Comment by jarekankowski** [comment]
@MarcinMoskala sure, I won't it wasn't easy so kudos for the PR! Please do let me know if you have any following questions to my comments (here or feel free to reach out to me on slack if there's anything specific you'd like to discuss).

Talking about DI - yes, 100% to do it in another PR. With all the great job you did with providing dependencies via constructor, will be relatively easy to introduce a DI library. Let’s just discuss later on which one should we use.

**Pull request merged**